src/AutoLaunchLinux.coffee

```diff
@@ -10,30 +10,39 @@ module.exports =
         file = @getDir()+opts.appName+'.desktop'
         return file
 
-    enable: (opts, cb) ->
-        file = @getFile(opts)
-
-        data = [
-          '[Desktop Entry]',
-          'Type=Application',
-          'Vestion=1.0',
-          'Name='+opts.appName,
-          'Comment=' + opts.appName + ' startup script',
-          'Exec=' + opts.appPath,
-          'StartupNotify=false',
-          'Terminal=false'
-        ].join('\n')
-
-        mkdirp.sync(@getDir())
-        fs.writeFileSync(file, data)
-        cb()
-
-    disable: (opts, cb) ->
-        file = @getFile(opts)
-        if fs.existsSync(file)
-            fs.unlinkSync(file)
-        cb()
-
-    isEnabled: (opts, cb) ->
-        file = @getFile(opts)
-        cb(fs.existsSync(file))
+    enable: (opts) ->
+        new Promise (resolve, reject) =>
+            file = @getFile(opts)
+
+            data = [
+                '[Desktop Entry]',
+                'Type=Application',
+                'Vestion=1.0',
+                'Name='+opts.appName,
+                'Comment=' + opts.appName + ' startup script',
+                'Exec=' + opts.appPath,
+                'StartupNotify=false',
+                'Terminal=false'
+            ].join('\n')
+
+            mkdirp.sync(@getDir())
+            fs.writeFile file, data, (err) ->
+                return reject if err?
+                resolve()
+
+    disable: (opts) ->
+        new Promise (resolve, reject) =>
+            file = @getFile(opts)
+
+            fs.stat file, (err) ->
+                return reject(err) if err?
+                fs.unlink file, (err) ->
+                    return reject if err?
+                    resolve()
+
+    isEnabled: (opts) ->
+        new Promise (resolve, reject) =>
+            file = @getFile(opts)
+
+            fs.stat file, (err, stat) ->
+                resolve(stat?)
```

Comment:
My concern here is how we differentiate between disable and error states. For instance in Linux implementation a `ENOENT` error means disable state, while `EACCES` indicates a real error.

The same applies for Windows but is even more difficult to tackle due to the lack of information provided by `winreg` lib. They've added support for capture error messages in [new versions](https://github.com/fresc81/node-winreg/pull/23), maybe we could update to latest version in order to get that info.

In Mac error and returned data are separated so there's nothing to change.

So I'm wondering how can we approach this... Should we differentiate error types? What are your thoughts?